### DISCUSSION TITLE
How would I obtain a list of all the files our that have a dependcey on a common.hcl file

### DISCUSSION THREAD

**Discussion by NickSpaghetti** [gruntwork-io/knowledge-base#719] [How do I]
Hey Gruntwork team,
we have a `common.hcl ` file at the root of our `infrastructure-live` that keeps track of all of our accounts under terragrunt that contains data that would normally have to be duplicated inside of multiple modules inputs.  
When that file gets updated, we want to run plan on, all the modules that use `common.hcl`.  Is there a Terragrunt command to give me a list of files that reference `common.hcl `. We are following project guide lines for [terraform-aws-service-catalog](https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/master/examples/for-production/infrastructure-live/_ci/scripts/deploy-infra.sh#L37). I noticed that if `$updated_folder` is equal to root we then echo a run-all suggestion and bail on plan/apply.   What is the reasons to not use plan-all or apply-all?  I know these commands are deprecated, and anything newer should use run-all.   My end goal is to obtain a list of files that have a dependency on `common.hcl` and run plan/apply in parallel inside of our github action with manual approvals on apply.  is there a recommended way to manage configuration files that are included in many `terragrunt.hcl `files across infrastructure-live?

---

<ins datetime="2023-05-26T18:47:12Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110205">Tracked in ticket #110205</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
since `common.hcl` is loaded through `rearead_terragrunt_config / find_in_parent_folders` like `read_terragrunt_config(find_in_parent_folders("common.hcl"))` I didn't found how to make Terragrunt to return list of dependencies in one line, however, approach an be separated in multiple steps: 

1. find all paths which reference `common.hcl`
```
# save list 
find "$(pwd)" -type f -name 'terragrunt.hcl' -exec grep -q 'common\.hcl' {} \; -exec dirname {} \; | xargs -I{} readlink -f {}
```

2. Run Terragrunt with `--terragrunt-strict-include --terragrunt-include-dir ...` and pass directories which refernce common.hcl:
```
terragrunt run-all validate  --terragrunt-non-interactive --terragrunt-strict-include --terragrunt-include-dir list_item_1 --terragrunt-include-dir list_item_2 --terragrunt-include-dir list_item_3
```

**Reply by NickSpaghetti** [reply]
Thanks for the reply,  I have modified step 1 to filter out .terragrunt-cache since it was picking that up.  Looking at the documentation I saw a flag for the command `run-all` [--terragrunt-modules-that-include](https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-modules-that-include).  When I try to use the command `terragrunt run-all plan --terragrunt-modules-that-include /Users/foo/infrastructure-live/common.hcl` the only output I get is the following with no plan being run.
```
INFO[0000] The stack at /Users/foo/infrastructure-live/github will be processed in the following order for command plan:
```
When I look at the debug logs I see that there are 3 terragrunt files found.  Same thing when I run the command from step 1 in your response.  However, no plan is ever ran.  I was wondering if I am missing something or if this is a bug?

**Reply by denis256** [reply]
Hi,
from what I found, `--terragrunt-modules-that-include` will match only modules which use `include` block, in `terraform-aws-service-catalog` is used multiple times `read_terragrunt_config` which is not matched but still add dependency to common hcl file
